src/context/alert.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import React, { createContext, FunctionComponent, useState } from "react";
+import {
+  AlertModal,
+  AlertModalProp
+} from "../components/AlertModal/AlertModal";
+
+const defaultAlertProp: AlertModalProp = {
+  alertType: "ERROR",
+  title: "",
+  description: "",
+  buttonTextType: "OK_CANCEL",
+  visible: false,
+  onOk: () => {},
+  onCancel: () => {},
+  onExit: () => {}
+};
+
+interface AlertModalContext {
+  showAlert: (props: AlertModalProp) => void;
+  clearAlert: () => void;
+}
+
+export const AlertModalContext = createContext<AlertModalContext>({
+  showAlert: () => null,
+  clearAlert: () => null
+});
+
+export const AlertModalContextProvider: FunctionComponent = ({ children }) => {
+  const [alertProps, setAlertProps] = useState<AlertModalProp>(
+    defaultAlertProp
+  );
+
+  const showAlert: AlertModalContext["showAlert"] = (props: AlertModalProp) => {
+    setAlertProps(props);
+  };
```

Comment:
Probably want to wrap these callbacks in `useCallback` so they don't cause unnecessary rerenders. Will need to take a closer look at performance in future

Reply:
Just thinking that this is a transient component. Is it necessary to use `useCallback`?

Reply:
Hmm, my main concern/hypothesis is when any of its parent context providers re-renders, the `AlertModalContext.Provider` re-renders unnecessarily since `showAlert` and `clearAlert` actually doesn't need to be a new reference on every render. But with that said, it's a hypothesis, will need to try and measure the performance of the context changes

Reply:
I tried to monitor to see if it re-renders unnecessarily. Didn't encounter it while doing some basic actions. I think we should monitor the perf first before introducing `useCallback` or `useMemo`. Maybe create a chore for perf testing/monitoring.
![image](https://user-images.githubusercontent.com/6116504/89368966-622fc300-d70f-11ea-9e21-4aa4fef13c51.png)